### DISCUSSION TITLE
Questions about the isio architecture

### DISCUSSION THREAD

**Discussion by eottabom** [istio/istio#51359] [Q&A] [answered]
Questions about the isio architecture

Up until the version of isio 1.4, the isiod consisted of pilot, mixer, Citadel, and Galley (each Pod)

Beginning with version of isio 1.5, it was integrated into a single module called isiod (run with isiod Pod)
https://istio.io/latest/news/releases/1.5.x/announcing-1.5/
https://istio.io/latest/blog/2020/istiod/

Mixer is defined from 1.5 (completely disappeared from 1.7), and the role of Mixer is

It was used to combine with logging, metric APIs, and APIs for billing, but it was removed for performance reasons,

integrated into Envoy.
([Redefining extensibility in proxies - introducing WebAssembly to Envoy and Istio](https://istio.io/latest/blog/2020/wasm-announce/))

Citadel is a module that performs security-related functions and provides access to services such as services and services ([Security](https://istio.io/latest/blog/2020/wasm-announce/))

Galley provides the configuration management services of isio (like changing to a structure applicable to kubenets.yaml → isio)

→ Currently, Citadel, Galley say only some code is built-in.

![istio](https://github.com/istio/istio/assets/37852701/eb40ce10-45e3-4205-aa2e-21f849e9ab06)

https://istio.io/latest/docs/ops/deployment/architecture/
![istio_architecture](https://github.com/istio/istio/assets/37852701/f75146a2-8305-45d2-b4d2-175d00750190)

The architecture in the istio docs seems to be rather confusing.

Because the pods for pilot, citadel, galley don't exist now and are built into code.

Am I right to understand?

**Comment by howardjohn** [comment, marked as answer]
There is now one component - Istiod. It fulfils the roll that Pilot+Citadel+Galley used to fill, by bundling those into one thing.

The docs probably shouldn't mention these legacy components anymore

**Reply by eottabom** [reply]
It seems that the document needs to be modified.

**Reply by craigbox** [reply]
Yep, and [#15233](https://github.com/istio/istio.io/issues/15233) is tracking that so I've closed your issue in favour of it.
